File: June-18/29-06-18/python/01-Motu-and-Patlu.py

for _ in range(int(input())):
    n = int(input())
    ar = [int(x) for x in input().split()]
    sum1 = 0
    sum2 = 0
    i = 0
    j = n - 1
    while(i <= j):
        if(sum1 <= (2*sum2)):
            while((sum1 <= 2*sum2) and i<=j):
                sum1+=ar[i];
                i+=1;
            
        
        elif(sum1>sum2):
            while((sum1>2*sum2) and j>=i):
                sum2+=ar[j];
                j-=1;
            
        
    a = i
    b = n - j - 1
    print(a,b)
    if(a > b):
        print("Motu")
    elif(a < b):
        print("Patlu")
    else:
        print("Tie")
    

File: June-18/29-06-18/c++/01-Motu-and-Patlu.cpp

#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n;
        cin >> n;
        int ar[n];
        for(int i = 0; i < n; i++)
        {
            cin >> ar[i];
        }
        
        
        
        long sum1 = 0,sum2 = 0;
        int i = 0, j = n - 1;
        
        
        while(i <= j)
        {
            if(sum1 <= (2*sum2))
            {
                while((sum1 <= 2*sum2) && i<=j)
                {
                    sum1+=ar[i];
                    i++;
                }
            }
            else
            if(sum1>sum2)
            {
                while((sum1>2*sum2)&&j>=i)
                {
                    sum2+=ar[j];
                    j--;
                }
            }
        }
        
        
        int a,b;
        a = i;
        b = n - j - 1;
        cout << a << " " << b << endl;
        if(a > b)
            cout << "Motu\n";
        else if(a < b)
            cout << "Patlu\n";
        else
            cout << "Tie\n";
        
    }
}
